tests/e2e/basic-ingest-redis-serving.py

```diff
@@ -539,10 +547,86 @@ def test_all_types_parquet_register_feature_set_success(client):
 @pytest.mark.timeout(600)
 @pytest.mark.run(order=41)
 def test_all_types_infer_register_ingest_file_success(client,
-    all_types_parquet_file):
+                                                      all_types_parquet_file):
     # Get feature set
     all_types_fs = client.get_feature_set(name="all_types_parquet")
 
     # Ingest user embedding data
     client.ingest(feature_set=all_types_fs, source=all_types_parquet_file,
                   force_update=True)
+
+
+# TODO: rewrite these using python SDK once the labels are implemented there
+class TestsBasedOnGrpc:
+    LAST_VERSION = 0
+    GRPC_CONNECTION_TIMEOUT = 3
+    LABEL_KEY = "my"
+    LABEL_VALUE = "label"
+
+    @pytest.fixture(scope="module")
+    def core_service_stub(self, core_url):
+        if core_url.endswith(":443"):
+            core_channel = grpc.secure_channel(
+                core_url, grpc.ssl_channel_credentials()
+            )
+        else:
+            core_channel = grpc.insecure_channel(core_url)
+
+        try:
+            grpc.channel_ready_future(core_channel).result(timeout=self.GRPC_CONNECTION_TIMEOUT)
+        except grpc.FutureTimeoutError:
+            raise ConnectionError(
+                f"Connection timed out while attempting to connect to Feast "
+                f"Core gRPC server {core_url} "
+            )
+        core_service_stub = CoreServiceStub(core_channel)
+        return core_service_stub
+
+    def apply_feature_set(self, core_service_stub, feature_set_proto):
+        try:
+            apply_fs_response = core_service_stub.ApplyFeatureSet(
+                CoreService_pb2.ApplyFeatureSetRequest(feature_set=feature_set_proto),
+                timeout=self.GRPC_CONNECTION_TIMEOUT,
+            )  # type: ApplyFeatureSetResponse
+        except grpc.RpcError as e:
+            raise grpc.RpcError(e.details())
+        return apply_fs_response.feature_set
+
+    def get_feature_set(self, core_service_stub, name, project):
+        try:
+            get_feature_set_response = core_service_stub.GetFeatureSet(
+                CoreService_pb2.GetFeatureSetRequest(
+                    project=project, name=name.strip(), version=self.LAST_VERSION
+                )
+            )  # type: GetFeatureSetResponse
+        except grpc.RpcError as e:
+            raise grpc.RpcError(e.details())
+        return get_feature_set_response.feature_set
+
+    @pytest.mark.timeout(45)
+    @pytest.mark.run(order=9)
```

Comment:
I think @woop uses these to group similar tests together (10- - so maybe your tests can start from 50?

Reply:
Perhaps this stuff can be tweaked when changing the tests to use SDK in #663